source/openpower/keys.rst

```diff
@@ -0,0 +1,160 @@
+
+If you have the correct credentials (the desired SSH private key), then you will be granted access to the server.
+
+.. note::
+
+  This description is very brief and not detailed enough to fully eclipse how SSH keys and the Diffie-Hellman exchange works. A more descriptive overview is linked in the section `above.`__
+  
+.. __: keys.html#ssh-key-behavior-and-management-on-openpower
+
+*However*, a common issue occurs **after rebuilding an instance**, regarding your local 'known_hosts' file, which stores a list of IP addresses and their corresponding DSA, RSA, or ECDSA host keys. 
+
+If you are recieving something similar to the error::
+
+  ECDSA host key for 140.211.123.456 has changed and you have requested strict checking.
+  Host key verification failed.
+
+then a simple command should fix the error::
+
+  ~$ ssh-keygen -f "<path/to>/.ssh/known_hosts" -R <instance_IP_address>
+  
+*Example:* ``~$ ssh-keygen -f "/home/scobeavs/.ssh/known_hosts" -R 140.211.123.456``
+
+This issue occurs because the client (the machine you're attempting to log on to the server through), is also attempting to authenticate the server, to keep the user (you) from connecting to a potentially dangerous imposter. This command removes the server from your 'known_hosts' file, allowing you to connect to the server as if it were brand new to you.
+
+Changing a Key From Inside an Instance
+--------------------------------------
+
+When we create our OpenPOWER OpenStack instance, we set up a key pair to create the machine with.
+
+.. note::
+
+  For the purposes of this documentation, this key pair will be known as the **Original Key Pair**.
+
+To allow for multiple SSH Keys to access this instance:
+  We need to add the contents of ``<our_key>.pub`` to ``~/.ssh/authorized_keys`` from the machine with the corresponding keyfile.
+  
+  To do so, we need to be certain we can ``ssh`` into our instance with:: 
+  
+      ~$ ssh <image_name>@<instance_IP_address>
+    
+  After we have tested that, we can ``Ctrl+D`` or ``logout`` out of the instance, and use ``ssh-copy-id`` to transfer ``<our_key>.pub`` to ``authorized_keys``, allowing multiple key pairs.
+  
+  - To **add** a key pair, retaining the original(s)::
+  
+      ~$ ssh-copy-id <path/to/our_key>.pub <image_name>@<instance_IP_address>
+      
+    *Example:* ``~$ ssh-copy-id ~/.ssh/id_rsa.pub ubuntu@140.311.133.05``
+    
+  - To **change** the key pair, removing all others::
+
+      ~$ cat <path/to/our_key>.pub | ssh <image_name>@<instance_IP_address> "mkdir -p ~/.ssh && cat > ~/.ssh/authorized_keys"
+    
+    *Example:* ``~$ cat ~/.ssh/id_rsa.pub | ssh debian@140.244.313.10 "mkdir...``    
+    
+Changing a Key From OpenStack Web GUI
+-------------------------------------
+
+As far as in our knowledge, there is currently no way to edit the **Original Key Pair** designated to the instance.
```

Comment:
The beginning of this sentence is wordy, It might look better shorter or even removed up to the comma.